Core/src/main/java/com/paypal/android/core/analytics/DeviceInspector.kt

```diff
@@ -0,0 +1,63 @@
+package com.paypal.android.core.analytics
+
+import android.content.Context
+import android.content.pm.PackageManager
+import android.os.Build
+import com.paypal.android.core.BuildConfig
+import com.paypal.android.core.analytics.models.DeviceData
+
+class DeviceInspector internal constructor(
+    private val clientSDKVersion: String,
+    private val sdkInt: Int,
+    private val deviceManufacturer: String,
+    private val deviceModel: String,
+    private val deviceProduct: String,
+    private val deviceFingerprint: String
+) {
+
+    constructor() : this(
+        clientSDKVersion = BuildConfig.PAYPAL_SDK_VERSION,
+        sdkInt = Build.VERSION.SDK_INT,
+        deviceManufacturer = Build.MANUFACTURER,
+        deviceModel = Build.MODEL,
+        deviceProduct = Build.PRODUCT,
+        deviceFingerprint = Build.FINGERPRINT
+    )
+
+    fun inspect(context: Context): DeviceData {
```

Comment:
`context` here sent as constructor param, saved in `WeakReference`, then we can get rid of `context` in `API`
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.native.ref/-weak-reference/

Reply:
We currently call `Context#getApplicationContext()` to store a long lived reference. Do we need a `WeakReference` here since we're storing a reference to the Application's context?

Reply:
Hey @kgangineni I removed this method in favor of a single `DeviceInspector` class.